content/en/docs/concepts/configuration/secret.md

```diff
@@ -72,732 +117,397 @@ merged into the `data` field. If a key appears in both the `data` and the
-        "creationTimestamp": "2021-07-01T07:30:59Z",
-        "name": "secret-tiger-docker",
-        "namespace": "default",
-        "resourceVersion": "566718",
-        "uid": "e15c1d7b-9071-4100-8681-f3a7a2ce89ca"
-    },
-    "type": "kubernetes.io/dockerconfigjson"
-}
+What will happen:
 
-```
+* the `username` key from `mysecret` is available to the container at the path
+  `/etc/foo/my-group/my-username` instead of at `/etc/foo/username`.
+* the `password` key from that Secret object is not projected.
 
-### Basic authentication Secret
+If `.spec.volumes[].secret.items` is used, only keys specified in `items` are projected.
+To consume all keys from the Secret, all of them must be listed in the `items` field.
 
-The `kubernetes.io/basic-auth` type is provided for storing credentials needed
-for basic authentication. When using this Secret type, the `data` field of the
-Secret must contain one of the following two keys:
+If you list keys explicitly, then all listed keys must exist in the corresponding Secret.
+Otherwise, the volume is not created.
 
-- `username`: the user name for authentication;
-- `password`: the password or token for authentication.
+#### Secret files permissions
 
-Both values for the above two keys are base64 encoded strings. You can, of
-course, provide the clear text content using the `stringData` for Secret
-creation.
+You can set the POSIX file access permission bits for a single Secret key.
+If you don't specify any permissions, `0644` is used by default.
+You can also set a default mode for the entire Secret volume and override per key if needed.
 
-The following YAML is an example config for a basic authentication Secret:
+For example, you can specify a default mode like this:
 
 ```yaml
 apiVersion: v1
-kind: Secret
+kind: Pod
 metadata:
-  name: secret-basic-auth
-type: kubernetes.io/basic-auth
-stringData:
-  username: admin
-  password: t0p-Secret
+  name: mypod
+spec:
+  containers:
+  - name: mypod
+    image: redis
+    volumeMounts:
+    - name: foo
+      mountPath: "/etc/foo"
+  volumes:
+  - name: foo
+    secret:
+      secretName: mysecret
+      defaultMode: 0400
 ```
 
-The basic authentication Secret type is provided only for user's convenience.
-You can create an `Opaque` for credentials used for basic authentication.
-However, using the builtin Secret type helps unify the formats of your credentials
-and the API server does verify if the required keys are provided in a Secret
-configuration.
+The secret is mounted on `/etc/foo`; all the files created by the
+secret volume mount have permission `0400`.
 
-### SSH authentication secrets
 
-The builtin type `kubernetes.io/ssh-auth` is provided for storing data used in
-SSH authentication. When using this Secret type, you will have to specify a
-`ssh-privatekey` key-value pair in the `data` (or `stringData`) field
-as the SSH credential to use.
+{{< note >}}
+If you're defining a Pod or a Pod template using JSON, beware that the JSON
```

Comment:
Not convinced here. Is “beware” unidiomatic?

Reply:
I don't know that we would use the words differently on either side of the pond, but I leave it to you. I just thought that "beware" implies some danger while "be aware" is just something that would be helpful to know. If you think this facts represents danger, then beware is fine.

Reply:
There is some danger here. People might specify the wrong `defaultMode` and leak a secret, because they thought that a number starting with `0` would be treated as octal.